Fix SubmitJobCommand returned error messages

zeineb opened the pull request.

sonarqube-activeeon commented:
SonarQube analysis reported 3 issues

1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [SubmitJobCommand.java#L63](https://github.com/ow2-proactive/scheduling/blob/42cbd3572452442d7545e03ed4979447a5554380/rest/rest-cli/src/main/java/org/ow2/proactive_grid_cloud_portal/cli/cmd/sched/SubmitJobCommand.java#L63): Replace this usage of System.out or System.err by a logger. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS106)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [SubmitJobCommand.java#L101](https://github.com/ow2-proactive/scheduling/blob/42cbd3572452442d7545e03ed4979447a5554380/rest/rest-cli/src/main/java/org/ow2/proactive_grid_cloud_portal/cli/cmd/sched/SubmitJobCommand.java#L101): Remove this unused method parameter "currentContext". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS1172)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [SubmitJobCommand.java#L158](https://github.com/ow2-proactive/scheduling/blob/42cbd3572452442d7545e03ed4979447a5554380/rest/rest-cli/src/main/java/org/ow2/proactive_grid_cloud_portal/cli/cmd/sched/SubmitJobCommand.java#L158): Move the "xml" string literal on the left side of this string comparison. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS1132)

sonarqube-activeeon commented:
SonarQube analysis reported 2 issues

1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [SubmitJobCommand.java#L101](https://github.com/ow2-proactive/scheduling/blob/2138207b53c5ff00f68ac62f44a9113790c92752/rest/rest-cli/src/main/java/org/ow2/proactive_grid_cloud_portal/cli/cmd/sched/SubmitJobCommand.java#L101): Remove this unused method parameter "currentContext". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS1172)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [SubmitJobCommand.java#L158](https://github.com/ow2-proactive/scheduling/blob/2138207b53c5ff00f68ac62f44a9113790c92752/rest/rest-cli/src/main/java/org/ow2/proactive_grid_cloud_portal/cli/cmd/sched/SubmitJobCommand.java#L158): Move the "xml" string literal on the left side of this string comparison. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS1132)

tobwiens commented on rest/rest-cli/src/main/java/org/ow2/proactive_grid_cloud_portal/cli/cmd/sched/SubmitJobCommand.java:
```diff
@@ -140,7 +142,7 @@ private Boolean isFileExisting(String pathname) {
 
     private Boolean isFilePathValid(String pathname) {
-        String regex = "^(\\/)?([^\\/\\\u0000]+(\\/)?)+\\.xml";
+        String regex = "^(?>[a-z]:)?(?>\\\\|\\/)?([^\\\\\\/?%*:|\\\"<>\\r\\n]+(?>\\\\|\\/)?)+$";
```
Can you rename this variable into something more expressive like correctPathnameRegex

zeineb replied:
@tobwiens I addressed your comment. Thanks a lot!

tobwiens commented on rest/rest-cli/src/main/java/org/ow2/proactive_grid_cloud_portal/cli/cmd/sched/SubmitJobCommand.java:
```diff
@@ -151,4 +153,12 @@ private Boolean isFilePathValid(String pathname) {
     }
 
+    private Boolean isXMLFile(String pathname) {
+        String extension = FilenameUtils.getExtension(pathname);
```
Can you rename this variable into something more expressive like fileExtension.

fviale commented on rest/rest-cli/src/main/java/org/ow2/proactive_grid_cloud_portal/cli/cmd/sched/SubmitJobCommand.java:
```diff
@@ -59,6 +60,7 @@
 
     public SubmitJobCommand(String... params) throws CLIException {
         if (params == null || params.length == 0) {
+            logger.log(Level.WARNING, "Error message: Workflow file path is required");
```
this should be a ERROR level message

fviale commented on rest/rest-cli/src/main/java/org/ow2/proactive_grid_cloud_portal/cli/cmd/sched/SubmitJobCommand.java:
```diff
@@ -97,17 +99,17 @@ public void execute(ApplicationContext currentContext) throws CLIException {
     private void validateFilePath(ApplicationContext currentContext) {
-        if (!isFilePathValid(pathname)) {
+
+        if (!isFilePathValid(pathname) || !isXMLFile(pathname)) {
```
by discarding all non-xml files, you break the possibility to submit a job archive as we can see in the following code:

if (APPLICATION_XML.getMimeType().equals(contentType)) {
                 jobId = currentContext.getRestClient().submitXml(currentContext.getSessionId(),
                                                                  new FileInputStream(jobFile),
                                                                  map(this.variables));
             } else {
                 jobId = currentContext.getRestClient().submitJobArchive(currentContext.getSessionId(),
                                                                         new FileInputStream(jobFile),
                                                                         map(this.variables));
             }


You should first investigate the purpose of submitJobArchive, if it's a valid or osbolete feature.

fviale replied:
After inspecting the server-side code, the submitting-a-job-archive feature has been removed a while ago. The client-side part is thus obsolete. For clarity, it should be removed (can be part of this pull request or another one)

fviale commented on rest/rest-cli/src/main/java/org/ow2/proactive_grid_cloud_portal/cli/cmd/sched/SubmitJobCommand.java:
```diff
@@ -76,6 +78,7 @@ public void execute(ApplicationContext currentContext) throws CLIException {
             validateFilePath(currentContext);
             File jobFile = new File(pathname);
             String contentType = URLConnection.getFileNameMap().getContentTypeFor(pathname);
+            System.out.println(contentType);
```
This seems to be a debug output, could you remove it ?

sonarqube-activeeon commented:
SonarQube analysis reported 7 issues

1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [SubmitJobCommand.java#L81](https://github.com/ow2-proactive/scheduling/blob/cd8b76c2b2935cb30fce9565098ebb789bb4ae02/rest/rest-cli/src/main/java/org/ow2/proactive_grid_cloud_portal/cli/cmd/sched/SubmitJobCommand.java#L81): Replace this usage of System.out or System.err by a logger. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS106)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [SubmitJobCommand.java#L102](https://github.com/ow2-proactive/scheduling/blob/cd8b76c2b2935cb30fce9565098ebb789bb4ae02/rest/rest-cli/src/main/java/org/ow2/proactive_grid_cloud_portal/cli/cmd/sched/SubmitJobCommand.java#L102): Remove this unused method parameter "currentContext". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS1172)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [SubmitJobCommand.java#L164](https://github.com/ow2-proactive/scheduling/blob/cd8b76c2b2935cb30fce9565098ebb789bb4ae02/rest/rest-cli/src/main/java/org/ow2/proactive_grid_cloud_portal/cli/cmd/sched/SubmitJobCommand.java#L164): Replace these toUpperCase()/toLowerCase() and equals() calls with a single equalsIgnoreCase() call. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS1157)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [SubmitJobCommand.java#L164](https://github.com/ow2-proactive/scheduling/blob/cd8b76c2b2935cb30fce9565098ebb789bb4ae02/rest/rest-cli/src/main/java/org/ow2/proactive_grid_cloud_portal/cli/cmd/sched/SubmitJobCommand.java#L164): Move the "application/xml" string literal on the left side of this string comparison. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS1132)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [SubmitJobCommand.java#L165](https://github.com/ow2-proactive/scheduling/blob/cd8b76c2b2935cb30fce9565098ebb789bb4ae02/rest/rest-cli/src/main/java/org/ow2/proactive_grid_cloud_portal/cli/cmd/sched/SubmitJobCommand.java#L165): Move the "application/zip" string literal on the left side of this string comparison. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS1132)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [SubmitJobCommand.java#L165](https://github.com/ow2-proactive/scheduling/blob/cd8b76c2b2935cb30fce9565098ebb789bb4ae02/rest/rest-cli/src/main/java/org/ow2/proactive_grid_cloud_portal/cli/cmd/sched/SubmitJobCommand.java#L165): Replace these toUpperCase()/toLowerCase() and equals() calls with a single equalsIgnoreCase() call. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS1157)
1. ![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) [SubmitJobCommand.java#L164](https://github.com/ow2-proactive/scheduling/blob/cd8b76c2b2935cb30fce9565098ebb789bb4ae02/rest/rest-cli/src/main/java/org/ow2/proactive_grid_cloud_portal/cli/cmd/sched/SubmitJobCommand.java#L164): Remove those useless parentheses. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AUselessParenthesesCheck)

sonarqube-activeeon commented:
SonarQube analysis reported 7 issues

1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [SubmitJobCommand.java#L81](https://github.com/ow2-proactive/scheduling/blob/cd8b76c2b2935cb30fce9565098ebb789bb4ae02/rest/rest-cli/src/main/java/org/ow2/proactive_grid_cloud_portal/cli/cmd/sched/SubmitJobCommand.java#L81): Replace this usage of System.out or System.err by a logger. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS106)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [SubmitJobCommand.java#L102](https://github.com/ow2-proactive/scheduling/blob/cd8b76c2b2935cb30fce9565098ebb789bb4ae02/rest/rest-cli/src/main/java/org/ow2/proactive_grid_cloud_portal/cli/cmd/sched/SubmitJobCommand.java#L102): Remove this unused method parameter "currentContext". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS1172)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [SubmitJobCommand.java#L164](https://github.com/ow2-proactive/scheduling/blob/cd8b76c2b2935cb30fce9565098ebb789bb4ae02/rest/rest-cli/src/main/java/org/ow2/proactive_grid_cloud_portal/cli/cmd/sched/SubmitJobCommand.java#L164): Move the "application/xml" string literal on the left side of this string comparison. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS1132)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [SubmitJobCommand.java#L164](https://github.com/ow2-proactive/scheduling/blob/cd8b76c2b2935cb30fce9565098ebb789bb4ae02/rest/rest-cli/src/main/java/org/ow2/proactive_grid_cloud_portal/cli/cmd/sched/SubmitJobCommand.java#L164): Replace these toUpperCase()/toLowerCase() and equals() calls with a single equalsIgnoreCase() call. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS1157)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [SubmitJobCommand.java#L165](https://github.com/ow2-proactive/scheduling/blob/cd8b76c2b2935cb30fce9565098ebb789bb4ae02/rest/rest-cli/src/main/java/org/ow2/proactive_grid_cloud_portal/cli/cmd/sched/SubmitJobCommand.java#L165): Replace these toUpperCase()/toLowerCase() and equals() calls with a single equalsIgnoreCase() call. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS1157)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [SubmitJobCommand.java#L165](https://github.com/ow2-proactive/scheduling/blob/cd8b76c2b2935cb30fce9565098ebb789bb4ae02/rest/rest-cli/src/main/java/org/ow2/proactive_grid_cloud_portal/cli/cmd/sched/SubmitJobCommand.java#L165): Move the "application/zip" string literal on the left side of this string comparison. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS1132)
1. ![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) [SubmitJobCommand.java#L164](https://github.com/ow2-proactive/scheduling/blob/cd8b76c2b2935cb30fce9565098ebb789bb4ae02/rest/rest-cli/src/main/java/org/ow2/proactive_grid_cloud_portal/cli/cmd/sched/SubmitJobCommand.java#L164): Remove those useless parentheses. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AUselessParenthesesCheck)

sonarqube-activeeon commented:
SonarQube analysis reported 7 issues

1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [SubmitJobCommand.java#L97](https://github.com/ow2-proactive/scheduling/blob/50e5ee9e566e3980daee1a32f2e0b646751fffb2/rest/rest-cli/src/main/java/org/ow2/proactive_grid_cloud_portal/cli/cmd/sched/SubmitJobCommand.java#L97): Remove this unused method parameter "currentContext". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS1172)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [SubmitJobCommand.java#L159](https://github.com/ow2-proactive/scheduling/blob/50e5ee9e566e3980daee1a32f2e0b646751fffb2/rest/rest-cli/src/main/java/org/ow2/proactive_grid_cloud_portal/cli/cmd/sched/SubmitJobCommand.java#L159): Move the "application/xml" string literal on the left side of this string comparison. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS1132)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [SubmitJobCommand.java#L159](https://github.com/ow2-proactive/scheduling/blob/50e5ee9e566e3980daee1a32f2e0b646751fffb2/rest/rest-cli/src/main/java/org/ow2/proactive_grid_cloud_portal/cli/cmd/sched/SubmitJobCommand.java#L159): Replace these toUpperCase()/toLowerCase() and equals() calls with a single equalsIgnoreCase() call. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS1157)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [SubmitJobCommand.java#L160](https://github.com/ow2-proactive/scheduling/blob/50e5ee9e566e3980daee1a32f2e0b646751fffb2/rest/rest-cli/src/main/java/org/ow2/proactive_grid_cloud_portal/cli/cmd/sched/SubmitJobCommand.java#L160): Replace these toUpperCase()/toLowerCase() and equals() calls with a single equalsIgnoreCase() call. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS1157)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [SubmitJobCommand.java#L160](https://github.com/ow2-proactive/scheduling/blob/50e5ee9e566e3980daee1a32f2e0b646751fffb2/rest/rest-cli/src/main/java/org/ow2/proactive_grid_cloud_portal/cli/cmd/sched/SubmitJobCommand.java#L160): Move the "application/zip" string literal on the left side of this string comparison. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS1132)
1. ![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) [SubmitJobCommand.java](https://github.com/ow2-proactive/scheduling/blob/50e5ee9e566e3980daee1a32f2e0b646751fffb2/rest/rest-cli/src/main/java/org/ow2/proactive_grid_cloud_portal/cli/cmd/sched/SubmitJobCommand.java): Replace all tab characters in this file by sequences of white-spaces. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS00105)
1. ![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) [SubmitJobCommand.java#L159](https://github.com/ow2-proactive/scheduling/blob/50e5ee9e566e3980daee1a32f2e0b646751fffb2/rest/rest-cli/src/main/java/org/ow2/proactive_grid_cloud_portal/cli/cmd/sched/SubmitJobCommand.java#L159): Remove those useless parentheses. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AUselessParenthesesCheck)

sophiesongge commented on rest/rest-cli/src/main/java/org/ow2/proactive_grid_cloud_portal/cli/cmd/sched/SubmitJobCommand.java:
```diff
@@ -46,109 +46,119 @@
+					|| contentType.toLowerCase().equals("application/zip"));
+		return false;
+	}
 
```
It's hard to see the differences in this file, because the form is changed. Could you please use ./gradlew spotlessApply ?

zeineb commented:
@sophiesongge I improved the code readability. Thanks a lot.

sonarqube-activeeon commented:
SonarQube analysis reported 6 issues

1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [SubmitJobCommand.java#L101](https://github.com/ow2-proactive/scheduling/blob/7dd1ebc44a26792c0d0d8133e1bca3429f3410e8/rest/rest-cli/src/main/java/org/ow2/proactive_grid_cloud_portal/cli/cmd/sched/SubmitJobCommand.java#L101): Remove this unused method parameter "currentContext". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS1172)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [SubmitJobCommand.java#L163](https://github.com/ow2-proactive/scheduling/blob/7dd1ebc44a26792c0d0d8133e1bca3429f3410e8/rest/rest-cli/src/main/java/org/ow2/proactive_grid_cloud_portal/cli/cmd/sched/SubmitJobCommand.java#L163): Move the "application/xml" string literal on the left side of this string comparison. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS1132)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [SubmitJobCommand.java#L163](https://github.com/ow2-proactive/scheduling/blob/7dd1ebc44a26792c0d0d8133e1bca3429f3410e8/rest/rest-cli/src/main/java/org/ow2/proactive_grid_cloud_portal/cli/cmd/sched/SubmitJobCommand.java#L163): Replace these toUpperCase()/toLowerCase() and equals() calls with a single equalsIgnoreCase() call. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS1157)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [SubmitJobCommand.java#L164](https://github.com/ow2-proactive/scheduling/blob/7dd1ebc44a26792c0d0d8133e1bca3429f3410e8/rest/rest-cli/src/main/java/org/ow2/proactive_grid_cloud_portal/cli/cmd/sched/SubmitJobCommand.java#L164): Replace these toUpperCase()/toLowerCase() and equals() calls with a single equalsIgnoreCase() call. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS1157)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [SubmitJobCommand.java#L164](https://github.com/ow2-proactive/scheduling/blob/7dd1ebc44a26792c0d0d8133e1bca3429f3410e8/rest/rest-cli/src/main/java/org/ow2/proactive_grid_cloud_portal/cli/cmd/sched/SubmitJobCommand.java#L164): Move the "application/zip" string literal on the left side of this string comparison. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS1132)
1. ![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) [SubmitJobCommand.java#L163](https://github.com/ow2-proactive/scheduling/blob/7dd1ebc44a26792c0d0d8133e1bca3429f3410e8/rest/rest-cli/src/main/java/org/ow2/proactive_grid_cloud_portal/cli/cmd/sched/SubmitJobCommand.java#L163): Remove those useless parentheses. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AUselessParenthesesCheck)

sonarqube-activeeon commented:
SonarQube analysis reported 6 issues

1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [SubmitJobCommand.java#L101](https://github.com/ow2-proactive/scheduling/blob/7dd1ebc44a26792c0d0d8133e1bca3429f3410e8/rest/rest-cli/src/main/java/org/ow2/proactive_grid_cloud_portal/cli/cmd/sched/SubmitJobCommand.java#L101): Remove this unused method parameter "currentContext". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS1172)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [SubmitJobCommand.java#L163](https://github.com/ow2-proactive/scheduling/blob/7dd1ebc44a26792c0d0d8133e1bca3429f3410e8/rest/rest-cli/src/main/java/org/ow2/proactive_grid_cloud_portal/cli/cmd/sched/SubmitJobCommand.java#L163): Replace these toUpperCase()/toLowerCase() and equals() calls with a single equalsIgnoreCase() call. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS1157)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [SubmitJobCommand.java#L163](https://github.com/ow2-proactive/scheduling/blob/7dd1ebc44a26792c0d0d8133e1bca3429f3410e8/rest/rest-cli/src/main/java/org/ow2/proactive_grid_cloud_portal/cli/cmd/sched/SubmitJobCommand.java#L163): Move the "application/xml" string literal on the left side of this string comparison. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS1132)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [SubmitJobCommand.java#L164](https://github.com/ow2-proactive/scheduling/blob/7dd1ebc44a26792c0d0d8133e1bca3429f3410e8/rest/rest-cli/src/main/java/org/ow2/proactive_grid_cloud_portal/cli/cmd/sched/SubmitJobCommand.java#L164): Move the "application/zip" string literal on the left side of this string comparison. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS1132)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [SubmitJobCommand.java#L164](https://github.com/ow2-proactive/scheduling/blob/7dd1ebc44a26792c0d0d8133e1bca3429f3410e8/rest/rest-cli/src/main/java/org/ow2/proactive_grid_cloud_portal/cli/cmd/sched/SubmitJobCommand.java#L164): Replace these toUpperCase()/toLowerCase() and equals() calls with a single equalsIgnoreCase() call. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS1157)
1. ![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) [SubmitJobCommand.java#L163](https://github.com/ow2-proactive/scheduling/blob/7dd1ebc44a26792c0d0d8133e1bca3429f3410e8/rest/rest-cli/src/main/java/org/ow2/proactive_grid_cloud_portal/cli/cmd/sched/SubmitJobCommand.java#L163): Remove those useless parentheses. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AUselessParenthesesCheck)

sonarqube-activeeon commented:
SonarQube analysis reported 6 issues

1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [SubmitJobCommand.java#L101](https://github.com/ow2-proactive/scheduling/blob/20ca61860cde0eff49e16e8ba84383563a3b8f84/rest/rest-cli/src/main/java/org/ow2/proactive_grid_cloud_portal/cli/cmd/sched/SubmitJobCommand.java#L101): Remove this unused method parameter "currentContext". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS1172)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [SubmitJobCommand.java#L163](https://github.com/ow2-proactive/scheduling/blob/20ca61860cde0eff49e16e8ba84383563a3b8f84/rest/rest-cli/src/main/java/org/ow2/proactive_grid_cloud_portal/cli/cmd/sched/SubmitJobCommand.java#L163): Move the "application/xml" string literal on the left side of this string comparison. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS1132)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [SubmitJobCommand.java#L163](https://github.com/ow2-proactive/scheduling/blob/20ca61860cde0eff49e16e8ba84383563a3b8f84/rest/rest-cli/src/main/java/org/ow2/proactive_grid_cloud_portal/cli/cmd/sched/SubmitJobCommand.java#L163): Replace these toUpperCase()/toLowerCase() and equals() calls with a single equalsIgnoreCase() call. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS1157)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [SubmitJobCommand.java#L164](https://github.com/ow2-proactive/scheduling/blob/20ca61860cde0eff49e16e8ba84383563a3b8f84/rest/rest-cli/src/main/java/org/ow2/proactive_grid_cloud_portal/cli/cmd/sched/SubmitJobCommand.java#L164): Replace these toUpperCase()/toLowerCase() and equals() calls with a single equalsIgnoreCase() call. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS1157)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [SubmitJobCommand.java#L164](https://github.com/ow2-proactive/scheduling/blob/20ca61860cde0eff49e16e8ba84383563a3b8f84/rest/rest-cli/src/main/java/org/ow2/proactive_grid_cloud_portal/cli/cmd/sched/SubmitJobCommand.java#L164): Move the "application/zip" string literal on the left side of this string comparison. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS1132)
1. ![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) [SubmitJobCommand.java#L163](https://github.com/ow2-proactive/scheduling/blob/20ca61860cde0eff49e16e8ba84383563a3b8f84/rest/rest-cli/src/main/java/org/ow2/proactive_grid_cloud_portal/cli/cmd/sched/SubmitJobCommand.java#L163): Remove those useless parentheses. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AUselessParenthesesCheck)

marcocast commented on rest/rest-cli/src/main/java/org/ow2/proactive_grid_cloud_portal/cli/cmd/sched/SubmitJobCommand.java:
```diff
@@ -59,6 +60,7 @@
 
     public SubmitJobCommand(String... params) throws CLIException {
         if (params == null || params.length == 0) {
+            logger.log(Level.SEVERE, "Error message: Workflow file path is required");
```
Maybe it should be log level FATAL

marcocast commented on rest/rest-cli/src/main/java/org/ow2/proactive_grid_cloud_portal/cli/cmd/sched/SubmitJobCommand.java:
```diff
@@ -96,20 +98,24 @@ public void execute(ApplicationContext currentContext) throws CLIException {
 
     }
 
-    private void validateFilePath(ApplicationContext currentContext) {
```
It is better to create a separate class that perform all these validation.

zeineb commented:
@marcocast I have picked the most severe level: (https://docs.oracle.com/javase/7/docs/api/java/util/logging/Level.html)
- SEVERE (highest value)
- WARNING
- INFO
- CONFIG
- FINE
- FINER
- FINEST (lowest value)

mbenguig commented on rest/rest-cli/src/main/java/org/ow2/proactive_grid_cloud_portal/cli/cmd/sched/SubmitJobCommand.java:
```diff
@@ -140,8 +146,8 @@ private Boolean isFileExisting(String pathname) {
     private Boolean isFilePathValid(String pathname) {
-        String regex = "^(\\/)?([^\\/\\\u0000]+(\\/)?)+\\.xml";
-        final Pattern pattern = Pattern.compile(regex);
+        String correctPathnameRegex = "^(?>[a-z]:)?(?>\\\\|\\/)?([^\\\\\\/?%*:|\\\"<>\\r\\n]+(?>\\\\|\\/)?)+$";
```
It is up to you to choose, either regex, or a more readable solution

public static boolean isValidPath(String path) {

            try {

                Paths.get(path);

            } catch (InvalidPathException | NullPointerException ex) {
                return false;
            }

            return true;
        }

mbenguig commented on rest/rest-cli/src/test/java/org/ow2/proactive_grid_cloud_portal/cli/cmd/sched/SubmitJobCommandTest.java:
```diff
@@ -126,15 +127,15 @@ public void testFilePathProvided() throws Exception {
 
         params = new String[1];
-        params[0] = "/file/path/non/valid/a.xm";
+        params[0] = "/src/test/java/config/c.xm";
```
Just need to know why there is no reference to System.getProperty("user.dir") as you did in testNonExistingFilePathProvided ..

sonarqube-activeeon commented:
SonarQube analysis reported 7 issues

1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [SubmitJobCommand.java#L102](https://github.com/ow2-proactive/scheduling/blob/d65ec0c6aa46c1a0f8a27f44c1f8e3975de7e3d4/rest/rest-cli/src/main/java/org/ow2/proactive_grid_cloud_portal/cli/cmd/sched/SubmitJobCommand.java#L102): Remove this unused method parameter "currentContext". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS1172)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [SubmitJobCommand.java#L153](https://github.com/ow2-proactive/scheduling/blob/d65ec0c6aa46c1a0f8a27f44c1f8e3975de7e3d4/rest/rest-cli/src/main/java/org/ow2/proactive_grid_cloud_portal/cli/cmd/sched/SubmitJobCommand.java#L153): Replace these toUpperCase()/toLowerCase() and equals() calls with a single equalsIgnoreCase() call. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS1157)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [SubmitJobCommand.java#L153](https://github.com/ow2-proactive/scheduling/blob/d65ec0c6aa46c1a0f8a27f44c1f8e3975de7e3d4/rest/rest-cli/src/main/java/org/ow2/proactive_grid_cloud_portal/cli/cmd/sched/SubmitJobCommand.java#L153): Move the "application/xml" string literal on the left side of this string comparison. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS1132)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [SubmitJobCommand.java#L154](https://github.com/ow2-proactive/scheduling/blob/d65ec0c6aa46c1a0f8a27f44c1f8e3975de7e3d4/rest/rest-cli/src/main/java/org/ow2/proactive_grid_cloud_portal/cli/cmd/sched/SubmitJobCommand.java#L154): Move the "application/zip" string literal on the left side of this string comparison. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS1132)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [SubmitJobCommand.java#L154](https://github.com/ow2-proactive/scheduling/blob/d65ec0c6aa46c1a0f8a27f44c1f8e3975de7e3d4/rest/rest-cli/src/main/java/org/ow2/proactive_grid_cloud_portal/cli/cmd/sched/SubmitJobCommand.java#L154): Replace these toUpperCase()/toLowerCase() and equals() calls with a single equalsIgnoreCase() call. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS1157)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [SubmitJobCommand.java#L164](https://github.com/ow2-proactive/scheduling/blob/d65ec0c6aa46c1a0f8a27f44c1f8e3975de7e3d4/rest/rest-cli/src/main/java/org/ow2/proactive_grid_cloud_portal/cli/cmd/sched/SubmitJobCommand.java#L164): Either log or rethrow this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS1166)
1. ![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) [SubmitJobCommand.java#L153](https://github.com/ow2-proactive/scheduling/blob/d65ec0c6aa46c1a0f8a27f44c1f8e3975de7e3d4/rest/rest-cli/src/main/java/org/ow2/proactive_grid_cloud_portal/cli/cmd/sched/SubmitJobCommand.java#L153): Remove those useless parentheses. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AUselessParenthesesCheck)

sophiesongge commented on rest/rest-cli/src/main/java/org/ow2/proactive_grid_cloud_portal/cli/cmd/sched/SubmitJobCommand.java:
```diff
@@ -98,7 +99,8 @@ public void execute(ApplicationContext currentContext) throws CLIException {
     }
 
-    private void validateFilePath(ApplicationContext currentContext) throws JobCreationRestException {
+    private void validateFilePath(ApplicationContext currentContext)
```
why do you need to throws 3 more exceptions here?

sophiesongge commented on rest/rest-cli/src/test/java/org/ow2/proactive_grid_cloud_portal/cli/cmd/sched/SubmitJobCommandTest.java:
```diff
@@ -105,12 +105,12 @@ public void init() {
         params[0] = new String();
-        params[0] = "";
+        params[0] = null;
 
```
Not sure I understand here, do you mean params[1] = null;?
Then one question to think about for params[0] = new String(), do you want a variable or a constant here?

zeineb commented:
@sophiesongge: I added exceptions for the validation of the workflow file path (well-formed XML, which is still a work in progress). I will remove them now and them in the next PR. 

@sophiesongge I need to have the params[0] a null object, not an empty String. params[0] is the workflow file path, and I added the check where the file path is not provided in the SubmitJob command.

sophiesongge commented:
@zeineb Yes, but the problem is:
         params[0] = new String();
         params[0] = null;
These two lines equals to: params[0] = null; 
and you never used params[1]

What I don't understand is why you created an empty String "" for params[0] first, then you make it equal to a null object.

zeineb commented:
@sophiesongge you are right, I removed the obsolete String param.

sonarqube-activeeon commented:
SonarQube analysis reported 7 issues

1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [SubmitJobCommand.java#L100](https://github.com/ow2-proactive/scheduling/blob/cefdee696bebf25dccf7a5e67a5e8cc4a0f59b41/rest/rest-cli/src/main/java/org/ow2/proactive_grid_cloud_portal/cli/cmd/sched/SubmitJobCommand.java#L100): Remove this unused method parameter "currentContext". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS1172)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [SubmitJobCommand.java#L150](https://github.com/ow2-proactive/scheduling/blob/cefdee696bebf25dccf7a5e67a5e8cc4a0f59b41/rest/rest-cli/src/main/java/org/ow2/proactive_grid_cloud_portal/cli/cmd/sched/SubmitJobCommand.java#L150): Move the "application/xml" string literal on the left side of this string comparison. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS1132)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [SubmitJobCommand.java#L150](https://github.com/ow2-proactive/scheduling/blob/cefdee696bebf25dccf7a5e67a5e8cc4a0f59b41/rest/rest-cli/src/main/java/org/ow2/proactive_grid_cloud_portal/cli/cmd/sched/SubmitJobCommand.java#L150): Replace these toUpperCase()/toLowerCase() and equals() calls with a single equalsIgnoreCase() call. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS1157)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [SubmitJobCommand.java#L151](https://github.com/ow2-proactive/scheduling/blob/cefdee696bebf25dccf7a5e67a5e8cc4a0f59b41/rest/rest-cli/src/main/java/org/ow2/proactive_grid_cloud_portal/cli/cmd/sched/SubmitJobCommand.java#L151): Move the "application/zip" string literal on the left side of this string comparison. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS1132)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [SubmitJobCommand.java#L151](https://github.com/ow2-proactive/scheduling/blob/cefdee696bebf25dccf7a5e67a5e8cc4a0f59b41/rest/rest-cli/src/main/java/org/ow2/proactive_grid_cloud_portal/cli/cmd/sched/SubmitJobCommand.java#L151): Replace these toUpperCase()/toLowerCase() and equals() calls with a single equalsIgnoreCase() call. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS1157)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [SubmitJobCommand.java#L161](https://github.com/ow2-proactive/scheduling/blob/cefdee696bebf25dccf7a5e67a5e8cc4a0f59b41/rest/rest-cli/src/main/java/org/ow2/proactive_grid_cloud_portal/cli/cmd/sched/SubmitJobCommand.java#L161): Either log or rethrow this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS1166)
1. ![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) [SubmitJobCommand.java#L150](https://github.com/ow2-proactive/scheduling/blob/cefdee696bebf25dccf7a5e67a5e8cc4a0f59b41/rest/rest-cli/src/main/java/org/ow2/proactive_grid_cloud_portal/cli/cmd/sched/SubmitJobCommand.java#L150): Remove those useless parentheses. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AUselessParenthesesCheck)

sonarqube-activeeon commented:
SonarQube analysis reported 7 issues

1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [SubmitJobCommand.java#L100](https://github.com/ow2-proactive/scheduling/blob/cefdee696bebf25dccf7a5e67a5e8cc4a0f59b41/rest/rest-cli/src/main/java/org/ow2/proactive_grid_cloud_portal/cli/cmd/sched/SubmitJobCommand.java#L100): Remove this unused method parameter "currentContext". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS1172)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [SubmitJobCommand.java#L150](https://github.com/ow2-proactive/scheduling/blob/cefdee696bebf25dccf7a5e67a5e8cc4a0f59b41/rest/rest-cli/src/main/java/org/ow2/proactive_grid_cloud_portal/cli/cmd/sched/SubmitJobCommand.java#L150): Replace these toUpperCase()/toLowerCase() and equals() calls with a single equalsIgnoreCase() call. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS1157)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [SubmitJobCommand.java#L150](https://github.com/ow2-proactive/scheduling/blob/cefdee696bebf25dccf7a5e67a5e8cc4a0f59b41/rest/rest-cli/src/main/java/org/ow2/proactive_grid_cloud_portal/cli/cmd/sched/SubmitJobCommand.java#L150): Move the "application/xml" string literal on the left side of this string comparison. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS1132)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [SubmitJobCommand.java#L151](https://github.com/ow2-proactive/scheduling/blob/cefdee696bebf25dccf7a5e67a5e8cc4a0f59b41/rest/rest-cli/src/main/java/org/ow2/proactive_grid_cloud_portal/cli/cmd/sched/SubmitJobCommand.java#L151): Replace these toUpperCase()/toLowerCase() and equals() calls with a single equalsIgnoreCase() call. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS1157)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [SubmitJobCommand.java#L151](https://github.com/ow2-proactive/scheduling/blob/cefdee696bebf25dccf7a5e67a5e8cc4a0f59b41/rest/rest-cli/src/main/java/org/ow2/proactive_grid_cloud_portal/cli/cmd/sched/SubmitJobCommand.java#L151): Move the "application/zip" string literal on the left side of this string comparison. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS1132)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [SubmitJobCommand.java#L161](https://github.com/ow2-proactive/scheduling/blob/cefdee696bebf25dccf7a5e67a5e8cc4a0f59b41/rest/rest-cli/src/main/java/org/ow2/proactive_grid_cloud_portal/cli/cmd/sched/SubmitJobCommand.java#L161): Either log or rethrow this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS1166)
1. ![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) [SubmitJobCommand.java#L150](https://github.com/ow2-proactive/scheduling/blob/cefdee696bebf25dccf7a5e67a5e8cc4a0f59b41/rest/rest-cli/src/main/java/org/ow2/proactive_grid_cloud_portal/cli/cmd/sched/SubmitJobCommand.java#L150): Remove those useless parentheses. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AUselessParenthesesCheck)

marcocast commented on rest/rest-cli/src/main/java/org/ow2/proactive_grid_cloud_portal/cli/cmd/sched/SubmitJobCommand.java:
```diff
@@ -96,20 +97,24 @@ public void execute(ApplicationContext currentContext) throws CLIException {
             throw new CLIException(REASON_FILE_EMPTY, String.format("'%s' is empty.", pathname));
         }
 
+        if (!isFileExisting(pathname)) {
```
you can move this check before isFileEmpty(pathname). and then only check if the file is empty. It will simplify the code
